openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -345,6 +346,7 @@ def get_schedules_with_target_date_by_bin_and_orgs(schedule_date_field, target_d
         'enrollment__user__profile',
         'enrollment__course',
     ).filter(
+        Q(enrollment__course__end__isnull=True) | Q(enrollment__course__end__gte=today),
```

Comment:
I'm not sure `target_date` would work here, because, in the case of `send_recurring_nudge`, it is today minus 3 or 10 days, and, in the case of `send_upgrade_reminder`, it is today plus 2 days. I agree that it would be useful to have the date from `--date` plumbed all the way down to this query, but we currently only send the `day_offset` and the `target_day_str` to the task, so I'm going to have to refactor the tasks to accept a `today_str` which would require making another task because of our blue-green deploys...

I also think this work would be better left for when we refactor the resolver like Cale suggested so that it is created at the end of each task, because we would also have to do the work to pass the current date to the tasks for that.

For now, I could reproduce the today date by doing `target_day - datetime.timedelta(days=day_offset)` in the tasks. It's pretty hacky, and I'd rather just pass the actual date, but seeing as we are going to majorly refactor this bit later I think it's okay?

Reply:
ou - good point. Let's leave this as is for now. Maybe with a note in a comment?